src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -2047,6 +2046,23 @@ string ConcatenateGlobalProperties(BuildRequestConfiguration configuration)
             }
         }
 
+        /// <summary>
+        /// Records the result to the current cache if its config isn't in the override cache.
+        /// </summary>
+        /// <param name="result">The result to potentially record in the current cache.</param>
+        internal void RecordResultToCurrentCacheIfConfigNotInOverrideCache(BuildResult result)
+        {
+            // Record these results to the current cache only if their config isn't in the
+            // override cache, which can happen if we are building in the project isolation mode
+            // ProjectIsolationMode.MessageUponIsolationViolation, and the received result was built by an
+            // isolation-violating dependency project.
+            if (_configCache is not ConfigCacheWithOverride
+                || !((ConfigCacheWithOverride)_configCache).HasConfigurationInOverrideCache(result.ConfigurationId))
```

Comment:
Nit: this can be simplified

```suggestion
            if (_configCache is not ConfigCacheWithOverride cacheWithOverride
                || !cacheWithOverride.HasConfigurationInOverrideCache(result.ConfigurationId))
```